### DISCUSSION TITLE
How to fetch faults like events

### DISCUSSION THREAD

**Discussion by adisai123** [vmware/govmomi#2803] [Community Help] [answered]
With this below method we can subscribe to events, how to subscribe to VimFault or Fault?

func (event.Manager).Events(ctx context.Context, objects []gvmtypes.ManagedObjectReference, pageSize int32, tail bool, force bool, f func(gvmtypes.ManagedObjectReference, []gvmtypes.BaseEvent) error, kind ...string) error,

**Comment by embano1** [comment]
Not sure if I'm understanding correctly, but you cannot "subscribe" to SOAP faults. These must be handled in your code for methods that return `error`. Related: https://github.com/vmware/govmomi/issues/2685

Can you please elaborate further on your use case?

**Reply by adisai123** [reply]
I am planning to capture vcenter events like dvsport blocked , any warning / info events.  I see there are no vMotion events, but there are faults available for the same. If you know how to capture in separate module.

**Comment by embano1** [comment, marked as answer]
Let me know if this question is resolved by our discussion in https://github.com/vmware/govmomi/discussions/2793. Sounds like this is a duplicate.

**Comment by embano1** [comment]
btw: I created an abstraction library on top of `govmomi` to simplify client and event handling: https://pkg.go.dev/github.com/embano1/vsphere/event

Example: https://github.com/embano1/vsphere/blob/main/example/main.go
